MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -1,231 +1,152 @@
-                     forKey:[groupId stringByAppendingString:batchId]];
+- (NSUInteger)countEntriesForTable:(NSString *)tableName where:(nullable NSString *)condition {
+  NSMutableString *countLogQuery = [NSMutableString stringWithFormat:@"SELECT COUNT(*) FROM \"%@\" ", tableName];
+  if (condition.length > 0) {
+    [countLogQuery appendFormat:@"WHERE %@", condition];
   }
-
-  // Load completed.
-  if (completion) {
-    completion([logs allValues], batchId);
-  }
-
-  // Return YES if more logs available.
-  return moreLogsAvailable;
+  NSArray<NSArray<NSNumber *> *> *result = [self executeSelectionQuery:countLogQuery];
+  return (result.count > 0) ? result[0][0].unsignedIntegerValue : 0;
 }
 
-#pragma mark - Delete logs
-
-- (NSArray<id<MSLog>> *)deleteLogsWithGroupId:(NSString *)groupId {
-  NSArray<id<MSLog>> *logs = [[self getLogsFromDBWithGroupId:groupId] allValues];
-
-  // Delete logs
-  [self deleteLogsFromDBWithColumnValue:groupId columnName:kMSGroupIdColumnName];
-
-  // Delete related batches.
-  for (NSString *batchKey in [self.batches allKeys]) {
-    if ([batchKey hasPrefix:groupId]) {
-      [self.batches removeObjectForKey:batchKey];
-    }
-  }
-  return logs;
-}
-
-- (void)deleteLogsWithBatchId:(NSString *)batchId groupId:(NSString *)groupId {
-
-  // Get log Ids.
-  NSString *batchIdKey = [groupId stringByAppendingString:batchId];
-  NSArray<NSString *> *Ids = self.batches[batchIdKey];
-
-  // Delete logs and associated batch.
-  if (Ids.count > 0) {
-    [self deleteLogsFromDBWithColumnValues:Ids columnName:kMSIdColumnName];
-    [self.batches removeObjectForKey:batchIdKey];
-  }
-}
-
-#pragma mark - DB selection
-
-- (NSDictionary<NSString *, id<MSLog>> *)getLogsFromDBWithGroupId:(NSString *)groupId {
-  NSString *selectLogQuery =
-      [NSString stringWithFormat:@"SELECT * FROM %@ WHERE %@ == '%@'", kMSLogTableName, kMSGroupIdColumnName, groupId];
-  return [self getLogsFromDBWithQuery:selectLogQuery];
-}
-
-- (NSDictionary<NSString *, id<MSLog>> *)getLogsFromDBWithGroupId:(NSString *)groupId limit:(NSUInteger)limit {
-
-  // Get ids from batches.
-  NSMutableArray<NSString *> *idsInBatches;
-  for (NSString *batchKey in [self.batches allKeys]) {
-    if ([batchKey hasPrefix:groupId]) {
-      [idsInBatches addObjectsFromArray:(NSArray<NSString *> * _Nonnull)self.batches[batchKey]];
+- (BOOL)executeNonSelectionQuery:(NSString *)query {
+  sqlite3 *db = NULL;
+  int result = SQLITE_OK;
+  result = sqlite3_open_v2([self.filePath UTF8String], &db, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE, NULL);
+  if (result == SQLITE_OK) {
+    char *errMsg;
+    result = sqlite3_exec(db, [query UTF8String], NULL, NULL, &errMsg);
+    if (result != SQLITE_OK) {
+      MSLogError([MSMobileCenter logTag], @"Query \"%@\" failed with error: %d - %@", query, result,
+                 [[NSString alloc] initWithUTF8String:errMsg]);
     }
+    sqlite3_close(db);
+  } else {
+    sqlite3_close(db);
```

Comment:
It's suggested on examples from the SQLite website itself so I believe it's there for a reason: https://sqlite.org/quickstart.html